x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -542,32 +542,107 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
 
     const newPolicy = (await this.get(soClient, id)) as PackagePolicy;
 
-    if (packagePolicy.package) {
-      await removeOldAssets({
-        soClient,
-        pkgName: packagePolicy.package.name,
-        currentVersion: packagePolicy.package.version,
-      });
+    await updatePackagePolicyVersion(packagePolicyUpdate, soClient, currentVersion);
 
-      if (packagePolicy.package.version !== currentVersion) {
-        const upgradeTelemetry: PackageUpdateEvent = {
-          packageName: packagePolicy.package.name,
-          currentVersion: currentVersion || 'unknown',
-          newVersion: packagePolicy.package.version,
-          status: 'success',
-          eventType: 'package-policy-upgrade' as UpdateEventType,
-        };
-        sendTelemetryEvents(
-          appContextService.getLogger(),
-          appContextService.getTelemetryEventsSender(),
-          upgradeTelemetry
-        );
-        appContextService.getLogger().info(`Package policy upgraded successfully`);
-        appContextService.getLogger().debug(JSON.stringify(upgradeTelemetry));
-      }
+    return newPolicy;
+  }
+
+  public async bulkUpdate(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient,
+    packagePolicyUpdates: Array<NewPackagePolicy & { version?: string; id: string }>,
+    options?: { user?: AuthenticatedUser; force?: boolean },
+    currentVersion?: string
```

Comment:
we currently have two different method for `upgrade` and `update` and Fleet I think it will make sense for now to have two bulk method two one for update one for upgrade

Reply:
@kpollich I am curious to have your though here and maybe more details on what is different in `update` and `upgrade`

Reply:
i wasn't able to find any bulk update method. I think upgrade seems different than update

Reply:
`update` is for making changes to the fields on a package policy, `upgrade` is for bumping the underlying version of the package and applying any new fields, variables, settings, etc from the old version to the new. 

I think they could both be handled in bulk without many issues.

Reply:
yeah that's exactly my purpose of adding bulkUpdate, it's meant for updating changing in the inputs

Reply:
@shahzad31 what I am not sure it's why we need this `currentVersion` here? if we need to do upgrades it's probably better to do it in a separate bulkUpgrade method? what do you think?

Reply:
Seems like we already pass `currentVersion` to the non-bulk `update` method so I assume we are just trying to be consistent with the APIs here. Looks like we only pass this `currentVersion` value in the internal usage of the `packagePolicyService.update` method here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/routes/package_policy/handlers.ts#L337

Maybe something we can look at removing in the future, but I'd rather just make the non-bulk + bulk implementations of this API match for now in this PR.